pkg/rancher-desktop/config/settings.ts

```diff
@@ -226,13 +226,18 @@ export function load(deploymentProfiles: DeploymentProfileType): Settings {
   } catch (err: any) {
     settings = clone(defaultSettings);
     if (err.code === 'ENOENT') {
```

Comment:
I think it is all correct now, but I still find the logic quite complex. Wouldn't this be equivalent and much clearer (pseudo-code):

```
     if (err.code === 'ENOENT') {
       _.merge(settings, deploymentProfiles.defaults);
       if (!Object.keys(deploymentProfiles.defaults).length && !Object.keys(deploymentProfiles.locked).length) {
         _isFirstRun = true;
       }
     }
     _.merge(settings, deploymentProfiles.locked);
     if (!_.has(settings, 'virtualMachine.memoryInGB') {
       # ... set default memory
     }
     # disable application updater for dev versions
     save(settings);
```

Reply:
I'm going to merge the change as-is (as it seems to be correct), but please simplify it in a separate PR!